com.unity.multiplayer.mlapi/Runtime/Core/NetworkObject.cs

```diff
@@ -798,5 +823,15 @@ internal static NetworkObject DeserializeSceneObject(NetworkBuffer objectStream,
 
             return networkObject;
         }
+
+        // Trigger the Interest system to do an update sweep on any Interest nodes
+        //  I am associated with
+        public void UpdateInterest()
+        {
+            foreach (var com in InterestNodes)
```

Comment:
I'd expect interest manager (or interest management system updated by networkmanager or networkupdateloop) to be self-contained and self-executing.
why are we updating interest nodes here on per networkobject level?
is this because of the concept shown above (interestsettingsoverride per networkobject-level)?

Reply:
Think of it this way...each prefab has **a** (not multiple, recently changed that btw to simplify) InterestNode, which itself is the root node of a "tree" of nodes **for that prefab**.  So whenever an instance of that prefab is instantiated, we go to that root node so we can populate its nodes so they know about that new instance (and ditto for de-spawn, and query).  There is one usability thing I need to work through however, which is while multiple prefabs may have the same root node (tree) if you have the same node appearing in 2 different trees bad things will happen.

Reply:
UpdateInterest is a little special.  With the other flows like querying, you indeed start from the central, self-contained InterestManager and traverse.  But Update-ing is all about giving you the ability to update just one node however (depending on how you implement Update) and whenever (depending on when you call Update)

Let's say you have a Grid node (and many other nodes) and you want to update just the grid node every 60 frames.  When you create your Grid node, you can hold on to the node reference and then trigger just its Update() call when desired